Library/Homebrew/test/test_cmd_testbot.rb

```diff
@@ -0,0 +1,44 @@
+require "testing_env"
+require "dev-cmd/test-bot"
+
+class TestbotCommandTests < Homebrew::TestCase
+  def with_environment(h)
+    old = Hash[h.keys.map { |k| [k, ENV[k]] }]
+    ENV.update h
+    begin
+      yield
+    ensure
+      ENV.update old
+    end
+  end
+
+  def test_resolve_test_tap
+    tap = Homebrew::resolve_test_tap
+    assert_equal tap, nil, "Should return nil if no tap slug provided"
+
+    slug = "spam/homebrew-eggs"
+    url = "https://github.com/#{slug}.git"
+    pairs = [
```

Comment:
`pairs` is a nice name for the structure of this thing, but what does it actually contain? (Subtle hint to maybe come up with a better variable name.)